# Data Exploration

- Team: Voting Demographics

- Data Source: IPUMS CPS (Current Population Survey) <https://cps.ipums.org/>

- Data Period: November 2024

- Target Outcome (Prediction Variables): Whether an individual voted / (Optional) Mode of voting (in person or mail)

- Features (Independent Variables): Age, Sex, Race, Educational attainment, Nativity, Region of residence, Duration of residence, Household type, Household income

In [2]:
import pandas as pd


In [4]:
COLSPECS = [(0,4), (4,9), (9,11), (11,21), (21,35), (35,37), (37,39), (39,40),
            (40,43), (43,45), (45,59), (59,74), (74,88), (88,90), (90,91),
            (91,94), (94,95), (95,97), (97, 98), (98,99), (99,101), (101,104),
            (104,105), (105,107), (107,109), (109,112)]

NAMES = ["YEAR", "SERIAL", "MONTH", "HWTFINL", "CPSID", "GQTYPE", "REGION",
         "METRO", "FAMINC", "PERNUM", "WTFINL", "CPSIDV", "CPSIDP", "AGE",
         "SEX", "RACE", "MARST", "FAMSIZE", "NCHILD", "NATIVITY", "EMPSTAT",
         "EDUC", "DIFFMOB", "VOTED", "VOTEHOW", "VOTERES"]

data = pd.read_fwf("data/cps_00007.dat", colspecs=COLSPECS, names=NAMES)
data.shape


(99063, 26)

### VOTED

Description: Voted in the most recent November election

Codes:
- 01: Did not vote  
- 02: Voted  
- 96: Refused  
- 97: Don't know  
- 98: No response  
- 99: Not in universe


In [5]:
data["VOTED"].value_counts()

VOTED
2     48387
99    33871
1     14708
97     1080
96     1017
Name: count, dtype: int64

### VOTE HOW

Description: Method of voting in the most recent November election

Codes:
- 01: In person
- 02: By mail  
- 96: Refused  
- 97: Don't know  
- 98: No response  
- 99: Not in universe

In [6]:
data["VOTEHOW"].value_counts()

VOTEHOW
99    50676
1     33886
2     14283
97      122
96       68
98       28
Name: count, dtype: int64

### AGE

In [7]:
data["AGE"].value_counts()

AGE
80    2672
85    2180
64    1371
61    1365
62    1362
      ... 
76     988
1      947
78     817
0      751
79     635
Name: count, Length: 82, dtype: int64

In [8]:
data["AGE"].describe()

count    99063.000000
mean        41.893007
std         23.635132
min          0.000000
25%         22.000000
50%         42.000000
75%         62.000000
max         85.000000
Name: AGE, dtype: float64

### SEX

Codes:
- 1: Male
- 2: Female
- 9: Not in universe

In [9]:
data["SEX"].value_counts()

SEX
2    50665
1    48398
Name: count, dtype: int64

### RACE

Codes:
- 100: White
- 200: Black
- 300: American Indian / Aleut / Eskimo
- 650: Asian or Pacific Islander
- 651: Asian only
- 652: Hawaiian / Pacific Islander only
- 700: Other (single) race, n.e.c.
- 801: White–Black
- 802: White–American Indian
- 803: White–Asian
- 804: White–Hawaiian / Pacific Islander
- 805: Black–American Indian
- 806: Black–Asian
- 807: Black–Hawaiian / Pacific Islander
- 808: American Indian–Asian
- 809: Asian–Hawaiian / Pacific Islander
- 810: White–Black–American Indian
- 811: White–Black–Asian
- 812: White–American Indian–Asian
- 813: White–Asian–Hawaiian / Pacific Islander
- 814: White–Black–American Indian–Asian
- 815: American Indian–Hawaiian / Pacific Islander
- 816: White–Black–Hawaiian / Pacific Islander
- 817: White–American Indian–Hawaiian / Pacific Islander
- 818: Black–American Indian–Asian
- 819: White–American Indian–Asian–Hawaiian / Pacific Islander
- 820: Two or three races, unspecified
- 830: Four or five races, unspecified
- 999: Blank

In [10]:
data["RACE"].value_counts()

RACE
100    79014
200     9547
651     6156
300     1253
801      880
802      664
803      564
652      441
804       98
809       83
813       80
805       80
810       74
806       40
811       24
812       14
817       11
816       10
808        8
815        7
807        6
814        4
830        3
818        1
820        1
Name: count, dtype: int64

### EDUCATIONAL ATTAINMENT

Codes:
- 000: NIU or no schooling
- 001: NIU or blank
- 002: None or preschool
- 010: Grades 1, 2, 3, or 4
- 011: Grade 1
- 012: Grade 2
- 013: Grade 3
- 014: Grade 4
- 020: Grades 5 or 6
- 021: Grade 5
- 022: Grade 6
- 030: Grades 7 or 8
- 031: Grade 7
- 032: Grade 8
- 040: Grade 9
- 050: Grade 10
- 060: Grade 11
- 070: Grade 12
- 071: 12th grade, no diploma
- 072: 12th grade, diploma unclear
- 073: High school diploma or equivalent
- 080: 1 year of college
- 081: Some college, but no degree
- 090: 2 years of college
- 091: Associate’s degree, occupational/vocational program
- 092: Associate’s degree, academic program
- 100: 3 years of college
- 110: 4 years of college
- 111: Bachelor’s degree
- 120: 5+ years of college
- 121: 5 years of college
- 122: 6+ years of college
- 123: Master’s degree
- 124: Professional school degree
- 125: Doctorate degree
- 999: Missing / Unknown

In [11]:
data["EDUC"].value_counts()

EDUC
73     23003
111    17944
1      16402
81     12376
123     8003
92      4802
91      3634
60      2349
50      2070
40      1838
125     1734
71      1330
30      1194
124     1086
20       723
10       356
2        219
Name: count, dtype: int64

### NATIVITY

Description: Foreign birthplace or parentage

Codes:
- 0: Unknown / Native-born
- 1: Both parents native-born
- 2: Father foreign-born, mother native-born
- 3: Mother foreign-born, father native-born
- 4: Both parents foreign-born
- 5: Foreign-born

In [12]:
data["NATIVITY"].value_counts()

NATIVITY
1    74190
5    14168
4     6351
2     2340
3     1909
0      105
Name: count, dtype: int64

### REGION OF RESIDENCE

Description: Region and division

Codes:
- 11: New England Division (Northeast Region)
- 12: Middle Atlantic Division (Northeast Region)
- 21: East North Central Division (Midwest Region)
- 22: West North Central Division (Midwest Region)
- 31: South Atlantic Division (South Region)
- 32: East South Central Division (South Region)
- 33: West South Central Division (South Region)
- 41: Mountain Division (West Region)
- 42: Pacific Division (West Region)
- 97: State not identified


In [13]:
data["REGION"].value_counts()

REGION
31    18503
42    14734
41    12581
33    11441
21    10064
22     9319
12     8304
11     7542
32     6575
Name: count, dtype: int64

### DURATION OF RESIDENCE

Description: Duration of residence at current address

Codes:
- 010: Less than 1 year
- 011: Less than 1 month
- 012: 1–6 months
- 013: 7–11 months
- 020: 1–2 years
- 030: 3 or more years
- 031: 3–4 years
- 032: 3–5 years
- 033: 5 years or longer
- 034: 6 years or longer
- 035: 6–9 years
- 036: 10 years or longer
- 996: Refused
- 997: Don’t know
- 998: No response
- 999: Not in universe

In [14]:
data["VOTERES"].value_counts()

VOTERES
33     40676
999    35412
31      9511
20      7502
10      5962
Name: count, dtype: int64

### HOUSEHOLD TYPE

Codes:
- 00: Non-interview household
- 01: Husband/wife primary family (neither Armed Forces)
- 02: Husband/wife primary family (either or both in Armed Forces)
- 03: Unmarried civilian male – primary family householder
- 04: Unmarried civilian female – primary family householder
- 05: Primary family householder – respondent in Armed Forces, unmarried
- 06: Civilian male primary individual
- 07: Civilian female primary individual
- 08: Primary individual householder – respondent in Armed Forces
- 09: Group quarters with family
- 10: Group quarters without family
- 99: Blank

In [15]:
data["GQTYPE"].value_counts()

GQTYPE
1     59379
4     13443
7      9398
6      9184
3      6573
2       940
10       58
5        43
8        32
9        13
Name: count, dtype: int64

### HOUSEHOLD INCOME

Description: Family income of householder

Codes:
- 100: Under $5,000
- 110: Under $1,000
- 111: Under $500
- 112: $500–999
- 120: $1,000–1,999
- 121: $1,000–1,499
- 122: $1,500–1,999
- 130: $2,000–2,999
- 131: $2,000–2,499
- 132: $2,500–2,999
- 140: $3,000–3,999
- 141: $3,000–3,499
- 142: $3,500–3,999
- 150: $4,000–4,999
- 200: $5,000–7,999
- 210: $5,000–7,499
- 220: $5,000–5,999
- 230: $6,000–7,999
- 231: $6,000–7,499
- 232: $6,000–6,999
- 233: $7,000–7,499
- 234: $7,000–7,999
- 300: $7,500–9,999
- 310: $7,500–7,999
- 320: $8,000–8,499
- 330: $8,500–8,999
- 340: $8,000–8,999
- 350: $9,000–9,999
- 400: $10,000–14,999
- 410: $10,000–10,999
- 420: $11,000–11,999
- 430: $10,000–12,499
- 440: $10,000–11,999
- 450: $12,000–12,999
- 460: $12,000–14,999
- 470: $12,500–14,999
- 480: $13,000–13,999
- 490: $14,000–14,999
- 500: $15,000–19,999
- 510: $15,000–15,999
- 520: $16,000–16,999
- 530: $17,000–17,999
- 540: $15,000–17,499
- 550: $17,500–19,999
- 560: $18,000–19,999
- 600: $20,000–24,999
- 700: $25,000–49,999
- 710: $25,000–29,999
- 720: $30,000–34,999
- 730: $35,000–39,999
- 740: $40,000–49,999
- 800: $50,000 and over
- 810: $50,000–74,999
- 820: $50,000–59,999
- 830: $60,000–74,999
- 840: $75,000 and over
- 841: $75,000–99,999
- 842: $100,000–149,999
- 843: $150,000 and over
- 995: Missing
- 996: Refused
- 997: Don’t know
- 999: Blank

In [16]:
data["FAMINC"].value_counts()

FAMINC
843    22464
842    17231
841    13400
830     9892
820     7020
740     6380
730     4187
720     3970
710     3156
600     3149
500     2108
100     1858
430     1564
470     1291
210      744
300      649
Name: count, dtype: int64

### METROPOLITAN

Description: Metropolitan and central/principal city status

Codes:
- 0: Not identified
- 1: Not in metropolitan area
- 2: In metropolitan area: In central/principal city
- 3: In metropolitan area: Not in central/principal city
- 4: In metropolitan area: Central/principal city status not identified
- 9: Missing/unknown

In [17]:
data["METRO"].value_counts()

METRO
3    40035
2    22799
1    18510
4    16603
0     1116
Name: count, dtype: int64

### MARITAL STATUS

Codes:
- 1: Married, spouse present
- 2: Married, spouse absent
- 3: Separated
- 4: Divorced
- 5: Widowed
- 6: Never married/single
- 9: NIU

In [18]:
data["MARST"].value_counts()

MARST
1    41476
6    25218
9    16402
4     8315
5     5288
3     1216
2     1148
Name: count, dtype: int64

### FAMILY SIZE

Description: Number of family members in household

Codes:
- 00: Missing
- 01-14: Number of members

In [19]:
data["FAMSIZE"].value_counts()

FAMSIZE
2     29546
4     18656
3     16929
1     15938
5     10030
6      4512
7      1743
8       992
9       324
10      190
11       77
12       72
14       28
13       26
Name: count, dtype: int64

### CHILDREN

Description: Number of children in household

Codes:
- 0-9: Number of children (9 is 9+)

In [20]:
data["NCHILD"].value_counts()

NCHILD
0    71923
1    12583
2     9320
3     3578
4     1137
5      334
6      117
7       36
8       20
9       15
Name: count, dtype: int64

### EMPLOYMENT STATUS

Codes:
- 00: NIU
- 01: Armed Forces
- 10: At work
- 12: Has job, not at work last week
- 20: Unemployed
- 21: Unemployed, experienced worker
- 22: Unemployed, new worker
- 30: Not in labor force
- 31: NILF, housework
- 32: NILF, unable to work
- 33: NILF, school
- 34: NILF, other
- 35: NILF, unpaid, lt 15 hours
- 36: NILF, retired

In [21]:
data["EMPSTAT"].value_counts()

EMPSTAT
10    45986
36    18713
0     16402
34    10581
32     3895
21     1662
12     1319
1       333
22      172
Name: count, dtype: int64

### DISABILITY LIMITING MOBILITY

Codes:
- 0: NIU
- 1: No mobility limitation
- 2: Has mobility limitation

In [22]:
data["DIFFMOB"].value_counts()

DIFFMOB
1    78329
0    16735
2     3999
Name: count, dtype: int64